bindings/pyroot/pythonizations/python/ROOT/_numbadeclare.py

```diff
@@ -65,15 +68,17 @@ def get_container_cpp_name(cpp_type: str) -> Union[str, None]:
             return None
         return CONTAINER_TYPES[container_type]["cpp_name"]
 
-    # Normalize input types by stripping ROOT and VecOps namespaces from input types
-    def normalize_typename(t):
+    def normalize_container_type(cpp_type: str) -> str:
         """
-        Remove ROOT:: and VecOps:: namespaces
+        Make sure all namespaces are included
+        e.g. "RVecI" -> "ROOT::RVecI"
         """
-        return t.replace("ROOT::", "").replace("VecOps::", "")
-
-    input_types = [normalize_typename(t) for t in input_types]
-    return_type = normalize_typename(return_type) if return_type is not None else None
+        container_type = get_container_type(cpp_type)
```

Comment:
```suggestion
        container_type = cpp_type if cpp_type in CONTAINER_TYPES else False
```
I think this `get_container_type()` function really obfuscates the code. I would be more explicit, then one has to jump around less to understand the code.

In general, I would avoid having too small functions that can be better inlined for the sake of readability. Also `is_container_type(cpp_type)` could be `cpp_type in CONTAINER_TYPES`. Then when reading the code I understand directly "okay it's checking if it's in the dictionary", and with `is_container_type(cpp_type)` I first assume there is more complicated logic that warrants a separate function.